### PR TITLE
Add Axostudio Yeti hotswap version

### PR DESCRIPTION
**Pull request by kb-elmo**
## Description

Adds support for the Yeti hotswap RGB PCB.
The PCB was commission designed by DZtech hence the differenct manufacturer in the config.h.
However it should ideally stay in the Axolstudio directory so that users can find it more easily (both the soldered and hotswap PCB are sold in the same groupbuy under the Axolstudio name).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/axolstudio/yeti/hotswap/hotswap.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
```
Comment (zvecr) [inline keyboards/axolstudio/yeti/hotswap/hotswap.c:123]:
```suggestion
```

File: keyboards/axolstudio/yeti/hotswap/config.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#    define DRIVER_COUNT 1
+#    define DRIVER_LED_TOTAL 64
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (zvecr) [inline keyboards/axolstudio/yeti/hotswap/config.h:96]:
```suggestion
```

File: keyboards/axolstudio/yeti/soldered/config.h [hunk 3]
```diff
@@ -40,3 +40,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
 #define DEBOUNCE 5
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (zvecr) [inline keyboards/axolstudio/yeti/soldered/config.h:46]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/axolstudio/yeti/hotswap/rules.mk [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+FORCE_NKRO = yes
```
Comment (fauxpark) [inline keyboards/axolstudio/yeti/hotswap/rules.mk:18]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by kb-elmo** [comment]
Bump :)

**Review by fauxpark — APPROVED** [review]

**Pull request merged**